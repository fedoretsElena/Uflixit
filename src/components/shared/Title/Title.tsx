import React, { Component } from "react";

interface IProps {
    title: string;
    length: number;
}

class Title extends Component<IProps> {
    render() {
        const { title, length } = this.props;

        return (
            <h4 className="mt-5 mb-3">
                {title}
                <span className="ml-2 badge badge-primary">{length}</span>
            </h4>
        );
    }

    shouldComponentUpdate(
        nextProps: Readonly<IProps>,
        nextState: Readonly<{}>,
        nextContext: any
    ): boolean {
        const { title, length } = this.props;

        if (title !== nextProps.title || length !== nextProps.length) {
            return true;
        }

        return false;
    }
}

export default Title;
