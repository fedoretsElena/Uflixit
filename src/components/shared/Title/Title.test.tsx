import Title from "./Title";

describe("Title component", () => {
    it("successful render component", () => {
        const component = shallow(<Title title={""} length={0} />);

        expect(component).toMatchSnapshot();
    });

    it("check is correct title in props", () => {
        const defaultTitle = "There is will be new title";

        const component = mount(<Title title={defaultTitle} length={0} />);

        expect(component.prop("title")).toEqual(defaultTitle);
    });

    it("render title correctly", () => {
        const defaultTitle = "There is will be new title";

        const component = mount(<Title title={defaultTitle} length={0} />);
        expect(component.find("h4")).toHaveLength(1);
    });
});
