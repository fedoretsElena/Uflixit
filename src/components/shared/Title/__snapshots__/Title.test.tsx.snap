// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component successful render component 1`] = `
<h4
  className="mt-5 mb-3"
>
  <span
    className="ml-2 badge badge-primary"
  >
    0
  </span>
</h4>
`;
